src/libclipper/src/rpc_service.cpp

```diff
@@ -269,8 +289,10 @@ void RPCService::receive_message(
 
         TaskExecutionThreadPool::create_queue(model, cur_replica_id);
         zmq_connection_id += 1;
+        document_receive_time(connection_id);
```

Comment:
Instead of calling this function in each "case" statement, let's make a single call on line 244 (after the creation of `connection_id`)